src/main.cpp

```diff
@@ -59,11 +59,11 @@ int main(int argc, char* argv[]) {
 
   basicFillRect = runner.getRenderer()->getBasicFillRect(NovelRT::GeoVector<float>(200, 200), NovelRT::RGBAConfig(0, 255, 255, 255), rectArgs);
 
-  runner.runOnUpdate([](const float delta) {
+  runner.runOnUpdate([](const double delta) {
```

Comment:
The update delta, however, is a double as that allows interpolation and other things. Die to clamping and other logic it is sufficiently small that rounding error is generally not a concern. It is also double instead of float to avoid these issues as well (and gives users the ability to do more precise updates when/if needed).

I can link to other articles/technical reports explaining why the delta should be double (rather than float) if desired.